comp/forwarder/defaultforwarder/forwarder.go

```diff
@@ -31,19 +33,46 @@ type provides struct {
 }
 
 func newForwarder(dep dependencies) provides {
-	return NewForwarder(dep.Config, dep.Log, dep.Lc, true, dep.Params)
+	options := creationOptions(dep.Params, dep.Config, dep.Log)
+	return NewForwarder(dep.Config, dep.Log, dep.Lc, true, options, dep.Params.useNoopForwarder)
+}
+
+func creationOptions(params Params, config config.Component, log log.Component) *Options {
+	var options *Options
+	if !params.withResolver {
+		options = NewOptions(config, log, getMultipleEndpoints(config, log))
+	} else {
+		keysPerDomain := getMultipleEndpoints(config, log)
+		options = NewOptionsWithResolvers(config, log, resolver.NewSingleDomainResolvers(keysPerDomain))
+	}
+	if disableAPIKeyChecking, ok := params.disableAPIKeyCheckingOverride.Get(); ok {
+		options.DisableAPIKeyChecking = disableAPIKeyChecking
+	}
+	for _, feature := range params.features {
+		options.EnabledFeatures = SetFeature(options.EnabledFeatures, feature)
+	}
```

Comment:
The `SetFeature` should be named `AppendFeature`, but not even sure it's necessary to have a function since it's only doing an `append`.

Overall, I'd probably do this differently: I'd add a `func (o *Options) SetFeatures(feature []Feature)` on the `Options` type directly, this function would do the for range to append all features.
Here, it would only be about: 

```
options.SetFeatures(params.features)
```

BTW what happened if a feature appear multiple times in `params.features` since it's a slice?

WDYT?

Reply:
I have updated this code to use the option pattern (`WithFeatures()`)

Reply:
Shouldn't you use it here?